### ISSUE TITLE
FCNPC_OnCreate and FCNPC_OnSpawn Callback Problem

### ISSUE THREAD

**Issue by AtlanticCityRoleplay** [ziggi/FCNPC#296]
FCNPC_OnCreate and FCNPC_OnSpawn callback is not Being Called...

Code: `https://github.com/ziggi/FCNPC/blob/master/sampsvr_files/filterscripts/traindrivers_pilots.pwn`

**Modification**

```
public FCNPC_OnCreate(npcid)
{
    print("Test Create");
    return 1;
}

public FCNPC_OnSpawn(npcid)
{
    print("Test Spawn");
    return 1;
}
```

**OUTPUT**

```
----------
Loaded log file: "server_log.txt".
----------

SA-MP Dedicated Server
----------------------
v0.3.7-R2, (C)2005-2015 SA-MP Team

[19:53:30] 
[19:53:30] Server Plugins
[19:53:30] --------------
[19:53:30]  Loading plugin: sscanf.dll
[19:53:31] 
[19:53:31]  ===============================
[19:53:31] 
[19:53:31]       sscanf plugin loaded.     
[19:53:31] 
[19:53:31]          Version: 2.13.8
[19:53:31] 
[19:53:31]    (c) 2022 Alex "Y_Less" Cole  
[19:53:31] 
[19:53:31]  ===============================
[19:53:31] 
[19:53:31]   Loaded.
[19:53:31]  Loading plugin: crashdetect.dll
[19:53:31]   CrashDetect plugin 4.21
[19:53:31]   Loaded.
[19:53:31]  Loading plugin: streamer.dll
[19:53:31] 

*** Streamer Plugin v2.9.6 by Incognito loaded ***

[19:53:31]   Loaded.
[19:53:31]  Loading plugin: FCNPC.dll
[19:53:31] 
[19:53:31] -------------------------------------------------
[19:53:31]      FCNPC - Fully Controllable NPC v2.0.5
[19:53:31]             Windows SA-MP 0.3.7 R2
[19:53:31]            Jul 22 2023 at 12:54:18
[19:53:31] 
[19:53:31]   Author:       OrMisicL (2013 - 2015)
[19:53:31]   Continued by: ziggi    (2016 - present)
[19:53:31] 
[19:53:31]   See full credits in the README.md file
[19:53:31] -------------------------------------------------
[19:53:31] 
[19:53:31] Loading...
[19:53:31] 
[19:53:31] -------------------------------------------------
[19:53:31]    ColAndreasv1.4.0
[19:53:31] 
[19:53:31]    Created By:
[19:53:31]      [uL]Chris42O
[19:53:31]      [uL]Slice
[19:53:31]      [uL]Pottus
[19:53:31] -------------------------------------------------
[19:53:31] 
[19:53:31] Loading...
[19:53:31] ColAndreas v1.4.0 Loaded.
[19:53:31]   Loaded.
[19:53:31]  Loading plugin: timerfix.dll
[19:53:31]   >> TimerFix v1.5 successfully loaded.
[19:53:31]   Loaded.
[19:53:31]  Loading plugin: pawncmd.dll
[19:53:31] [Pawn.CMD] plugin v3.4.0 loading...
[19:53:31] [Pawn.CMD] 

    | Pawn.CMD 3.4.0 | 2016 - 2023
    |--------------------------------
    | Author and maintainer: katursis


    | Compiled: Feb 12 2023 at 18:12:31
    |--------------------------------------------------------------
    | Repository: https://github.com/katursis/Pawn.CMD

[19:53:31]   Loaded.
[19:53:31]  Loading plugin: SKY.dll
[19:53:31] 
[19:53:31]  ===============================
[19:53:31]    
[19:53:31]    < SKY - 2.3.0 >
[19:53:31]    
[19:53:31]    (c) 2008 - Present | YSF Maintainers
[19:53:31]    (c) 2015 | Oscar "Slice" Broman
[19:53:31]    
[19:53:31]    Server Version: 0.3.7 R2
[19:53:31]    Operating System: Windows
[19:53:31]    Built on: Sep 27 2021 at 23:19:52
[19:53:31]    
[19:53:31]  ===============================
[19:53:31] 
[19:53:31]   Loaded.
[19:53:31]  Loading plugin: mysql.dll
[19:53:31]  >> plugin.mysql: R39-6 successfully loaded.
[19:53:31]   Loaded.
[19:53:31]  Loading plugin: discord-connector.dll
[19:53:36]  >> plugin.dc-connector: v0.3.5 successfully loaded.
[19:53:36]   Loaded.
[19:53:36]  Loading plugin: bcrypt-samp.dll
[19:53:36]   plugin.bcrypt v2.2.3 was loaded.
[19:53:36]   plugin.bcrypt: 8 cores detected, 7 threads will be used.
[19:53:36]   Loaded.
[19:53:36]  Loading plugin: sampvoice.dll
[19:53:36] [sv:dbg:network:init] : module initializing...
[19:53:36] [dbg:raknet:init] : module initializing...
[19:53:36] [dbg:raknet:init] : module initialized
[19:53:36] [sv:dbg:network:init] : module initialized
[19:53:36] [sv:dbg:main:Load] : creating 8 work threads...
[19:53:36]  -------------------------------------------    
[19:53:36]    ___                __   __    _              
[19:53:36]   / __| __ _ _ __  _ _\ \ / /__ (_) __ ___    
[19:53:36]   \__ \/ _` | '  \| '_ \   / _ \| |/ _/ -_)
[19:53:36]   |___/\__,_|_|_|_| .__/\_/\___/|_|\__\___|
[19:53:36]                   |_|                           
[19:53:36]  -------------------------------------------    
[19:53:36]            SampVoice by MOR loaded              
[19:53:36]  -------------------------------------------    
[19:53:36]   Loaded.
[19:53:36]  Loading plugin: nativechecker.dll
[19:53:36]   Loaded.
[19:53:36]  Loaded 12 plugins.

[19:53:36] 
[19:53:36] Filterscripts
[19:53:36] ---------------
[19:53:36]   Loading filterscript 'android-check.amx'...
[19:53:52] Loaded collision data.
[19:53:52] [sv:dbg:main:AmxLoad] : net game pointer (value:02419D70) received
[19:53:52] [sv:dbg:network:bind] : voice server running on port 64908
[19:53:52] Android check has been successfully loaded.
[19:53:52]   Loading filterscript 'acrp_map.amx'...
[19:53:52]   Loading filterscript 'acrp_main.amx'...
[19:53:52]   Loading filterscript 'acrp_store.amx'...
[19:53:52]   Loading filterscript 'acrp_dealership.amx'...
[19:53:52]   Loading filterscript 'xwar.amx'...
[19:53:52]   Loading filterscript 'npc_record.amx'...
[19:53:52]   Loading filterscript 'acrp_npc.amx'...
[19:53:52] [npc:join] TrainDriverLS has joined the server (998:127.0.0.1)
[19:53:52] [npc:join] TrainDriverSF has joined the server (997:127.0.0.1)
[19:53:52] [npc:join] TrainDriverLV has joined the server (996:127.0.0.1)
[19:53:52] [npc:join] PilotLS has joined the server (995:127.0.0.1)
[19:53:52] [npc:join] PilotSF has joined the server (994:127.0.0.1)
[19:53:52] [npc:join] PilotLV has joined the server (993:127.0.0.1)
[19:53:52] [npc:join] BusDriverLS has joined the server (992:127.0.0.1)
[19:53:52] [npc:join] BusDriverLV has joined the server (991:127.0.0.1)
[19:53:52] Train Driver LS [998]
[19:53:52] Train Driver SF [997]
[19:53:52] Train Driver LV [996]
[19:53:52] Pilot LS [995]
[19:53:52] Pilot SF [994]
[19:53:52] Pilot LV [993]
[19:53:52] Bus Driver LS [992]
[19:53:52] Bus Driver SF [991]
[19:53:52]   Loaded 8 filterscripts.

[19:53:52] *** Streamer Plugin: The plugin version (0x295) is older than the include file version (0x296) for this script. The plugin might need to be updated to the latest version.
[19:53:52] MapFix by Nexius v4.3.3 loaded (include version).
[19:53:52]  
[19:53:52] --------------------------------------
[19:53:52]  Anticheat Nex-AC loaded!
[19:53:52]  Anticheat version: 1.9.62
[19:53:52]  Author: Nexius
[19:53:52] --------------------------------------

[19:53:52] [LoadAntiCheat] Loading data from database...
[19:53:52] Loading Map.
[19:53:53] Loaded Map.                             
[19:53:53] Atlantic RP
[19:53:53] 148 objects loaded.
[19:53:53] Number of vehicle models: 7
[19:53:53] [Script] 18 atms loaded
[19:53:53] [Script] 3 houses loaded.
[19:53:53] [Script] 3 garages loaded.
[19:53:53] [Script] 19 Public Garage loaded
[19:53:53] [Script] 3 Petrol Station loaded
[19:53:53] [Script] 16 Oils loaded
[19:53:53] [Script] 23 businesses loaded.
[19:53:53] [Script] 37 entrances loaded.
[19:53:53] [Script] 24 factions loaded.
[19:53:53] [Script] 0 lands loaded.
[19:53:53] [Script] 13 gangs loaded.
[19:53:53] [Script] 4 points loaded.
[19:53:53] [Script] 4 turfs loaded.
[19:53:53] [ANTICHEAT]: Anti-cheat settings loaded successfully (loaded: 53). Time: 0 ??.
[19:53:53] [Server] Atlantic City Server loaded Successfully
[19:53:53] [Owner] Prince of India
[19:53:53] [Developer] Karthi Genius
[19:53:53] [Exclusive Mapper] RioFTW
```

**REFERENCE**

#200 Already Found a Issue and They have used Timer to Fix...,

Is it the only way to fix it by...? @ziggi

**Comment by ziggi** [comment]
No reason to duplicate the issue

**Comment by AtlanticCityRoleplay** [comment]
> No reason to duplicate the issue

Sry for that, i had a doubt regards that.

First Time Using FCNPC

**ANOTHER ISSUE**

While loading windows version, its showing newer version of plug-in available in 2.0.6 release

**Comment by ziggi** [comment]
> While loading windows version, its showing newer version of plug-in available in 2.0.6 release

Update to 2.0.7.

> Kindly update the Wiki

The project is not actively developed it's under the maintenance, you can update it by yourself if you want.

**Comment by AtlanticCityRoleplay** [comment]
> > While loading windows version, its showing newer version of plug-in available in 2.0.6 release
> 
> Update to 2.0.7.
> 
> > Kindly update the Wiki
> 
> The project is not actively developed it's under the maintenance, you can update it by yourself if you want.

Thanks for the Reply
